src/coreclr/jit/llvm.cpp

```diff
@@ -1387,6 +1418,37 @@ void Llvm::buildCnsLng(GenTree* node)
     mapGenTreeToValue(node, _builder.getInt64(node->AsLngCon()->LngValue()));
 }
 
+llvm::Value* Llvm::buildFieldList(GenTreeFieldList* fieldList,
+                                  Type*             llvmType)
+{
+    if (!fieldList->TypeIs(TYP_STRUCT))
+    {
+        failFunctionCompilation();
+    }
+
+    if (llvmType->isStructTy())
+    {
+        Value*   alloca     = _builder.CreateAlloca(llvmType);
+        Value*   llvmStruct = _builder.CreateLoad(alloca);
+        unsigned i          = 0;
+
+        assert(fieldList->Uses().IsSorted());
+
+        for (GenTreeFieldList::Use& use : fieldList->Uses())
+        {
+            _builder.CreateInsertValue(llvmStruct, consumeValue(use.GetNode(), llvmType->getStructElementType(i)), i);
+            i++;
+        }
```

Comment:
The field list is supposed to not be tied to the struct type, it should be built based only on the field types and offsets (`use.GetType` + `use.GetOffset`) stored inside the uses.

Reply:
Take the alloca, add the `use.GetOffset` and do the stores directly, i.e. plain LLVM stores, not `CreateInsertValue` ?

Reply:
Yep.

Reply:
Added a commit for this.  I'm not certain it is compiling any additional methods due to other `failFunctionCompilation`s but this commit, should get things green again.

I sneaked in a helper function when creating geps: if the offset is 0 the original address is returned, as I was getting annoyed at looking at `gep addr, 0` in the LLVM .  There is still a common occurrence of that for the shadow stack when calling a method, but that's a bit different so I'll try to refrain from doing that here.